### ISSUE TITLE
FR: Book description - improved formatting (e.g. left aligned text, or keep html formatting)

### ISSUE THREAD

**Issue by agk29** [koreader/koreader#10692]
The book description (from Book information in the reader menu) has fixed justified text and font size, and removes all html formatting. I currently export all of my books with extra information in the comment metadata (e.g. date modified in calibre, read status, word count, etc) so there are several short lines of text that look very odd with justified text (huge gaps between words). I also have several omnibus epubs where the comment metadata has descriptions for each book in the omnibus separated by a horizontal line \<hr>, but this doesn't appear in the description as it is converted to plain text and the descriptions become hard to parse. Any further options for formatting this text would be greatly appreciated

- Is it possible to keep the html formatting rather than converting to plain text? From a quick look at the code in filemanagerbookinfo.lua it uses htmlToPlainTextIfHtml() - is this necessary?
- If it is necessary and it must be in plain text format, can there be an option to NOT justify the text and have it left aligned instead?
- Currently the paragraphs of text are formatted with an indent, can this be changed to new line?
- Variable font size would also be nice :)

I would help out if I could but I have no experience with lua!

**Comment by poire-z** [comment]
I don't think we'd want to add such options for something of as little use as the Book description (even if _you_ use it for more - it is usually something one reads quickly and forgets, and then reads the book :) ), or care about making it rendered as proper html.
But newlines `<br/>` or paragraphs should get their last (or single) line left aligned.
If you don't, it may be an issue with our HtmlToPlainText stuff (depending on your HTML). (Or may be it was too complicated to fight single or double `<br/>` and avoiding blank line - and we didn't spend much time on that.)
Can you share one (or more if results expose different issues) of such of your EPUBs (just an extract, as EPUB, with the .opf and at least one html file kept so was can use it for testing.

We have one option in `Dict settings> [ ]  Justify text` - I think we use its value in other contexts that Dict lookup results too when we display bit of text. Dunno if we could also use it here.
I guess we assumed justified text is the preferred look when reading book related content :)

**Comment by hius07** [comment]
Not very pretty, but we can add Justify/Unjustify action to the long-pressing on the Find button.
(It already has a long-pressing action in the find-next mode)

**Comment by poire-z** [comment]
```html
<p><b>Popularity:</b> Rel. Unknown (8141 reviews) <b>Goodreads Rating:</b> 4.14 </p> <p> <b>Word count:</b> 191151 <b>Readability:</b> 11 (High School) </p> <p> <b>First published:</b> May 1980 Date added:</b> 12 Nov 2018 </p> <p><b>Genres/Moods:</b> Sci-Fi, Adventurous, Challenging, Slow-Paced, Spec-Fic</p> <p><b>Tags:</b> Space Opera, Space, Series, Novels, Hard Sci-Fi, Fiction, Fant, Aliens</p> <p><b>Books in Series:</b> 2</p> <hr> <div>
<p>Anthology containing:</p>
<p style="font-weight: bold">Dragon's Egg</p>
<p><strong>&acirc;In science fiction there is only a handful of books that stretch the mind&acirc;and this is one of them. &acirc;&acirc;Arthur C. Clarke</strong>In a moving story of sacrifice atriumph, human scientists establish a relationship with intelligent lifeforms&acirc;the cheela&acirc;living on Dragon&acirc;s Egg, a neutron star where one Earth hour is equivalent to hundreds oheir years. The cheela culturally evolve from savagery to the discovery of science, and for a brief time, men are their diligent teachers.</p>
<hr>
<p style="font-weight: bold">Starquake</p>
<p>Starquake, the sequel to Dragons Egg, takes place on the surface of a neutron star. The gravity is 67 billion Earth gravities. The native cheela, the size of sesame seeds, live a million timeaster than their human friends in orbit. After a starquake, the humans have only one day to save the remains of cheela civilization from extinction.</p>
<p></p>
</div>

<!-- when converted to plain text (leading spaces are actually tabs -->
<pre>
 Popularity: Rel. Unknown (8141 reviews) Goodreads Rating: 4.14
 Word count: 191151 Readability: 11 (High School)
 First published: May 1980 Date added: 12 Nov 2018
 Genres/Moods: Sci-Fi, Adventurous, Challenging, Slow-Paced, Spec-Fic
 Tags: Space Opera, Space, Series, Novels, Hard Sci-Fi, Fiction, Fantasy, Aliens
 Books in Series: 2

 Anthology containing:
Dragon's Egg
 âIn science fiction there is only a handful of books that stretch the mindâand this is one of them. ââArthur C. ClarkeIn a moving story of sacrifice and triumph, human scientists establish a reionship with intelligent lifeformsâthe cheelaâliving on Dragonâs Egg, a neutron star where one Earth hour is equivalent to hundreds of their years. The cheela culturally evolve from savagery to  discovery of science, and for a brief time, men are their diligent teachers.
Starquake
 Starquake, the sequel to Dragons Egg, takes place on the surface of a neutron star. The gravity is 67 billion Earth gravities. The native cheela, the size of sesame seeds, live a million times ter than their human friends in orbit. After a starquake, the humans have only one day to save the remains of cheela civilization from extinction.
</pre>
```

Current result:
![image](https://github.com/koreader/koreader/assets/24273478/bf964ebc-47cb-46e0-8cd4-e28f28e76ad2)
Contrary to what what was said (or what I understood :), single and last lines of paragraphs get left aligned.

TextViewer with `justified=false`:
![image](https://github.com/koreader/koreader/assets/24273478/0597d62a-3366-47a6-9d14-cf68bb54079b)
I don't think no-justification makes it significantly better - at least, with justification, you can guess the next line is a continuation of previous one.

Our HTML to plain text code:
https://github.com/koreader/koreader/blob/3e43a21cf268e05f122a9493016c4c343229a92d/frontend/util.lua#L1140-L1168
(`\t` gets replaced with the width of 8 spaces when rendering the text.)

Without the `\t` we use at start of paragraphs to make the indentation:
![image](https://github.com/koreader/koreader/assets/24273478/f4c182e2-b003-4e59-86c4-849cbb3cc31d)

**Comment by poire-z** [comment]
No, you get the same. You're just having a larger font size than I do, less words per lines, and so more paragraphs get to be multiline, and so you see more justified lines than I do.
And less words per line means less spaces between them, so each space gets the added width to ensure justification, so these large ugly gaps.

**Comment by poire-z** [comment]
> I also like how it looks when \t is removed from the plaintext - would there be any negative repercussions if I just edit that line in the koreader/frontend/util.lua file in my /.adds folder?

I think it shouldn't hurt.
You don't need to edit our shipped util.lua, which would get overwritten on each update.
You could just [user patch](https://github.com/koreader/koreader/wiki/User-patches) it with something like:
`2-util-htmlToPlainText-no-tab.lua`
```lua
local util = require("util")

local htmlToPlainText_orig = util.htmlToPlainText
util.htmlToPlainText = function(text)
    text = htmlToPlainText_orig(text)
    text = text:gsub("\t", "")
    return text
end
```

**Comment by poire-z** [comment]
Dunno if we could (in our non user-patch code) handle `<hr>` and replace it with `----` - might need to handle multiple `\n` before/after whether added by our `<hr>` replacement or by other stuff.
(Don't really want to think about it right now :)

**Comment by Frenzie** [comment]
Sometimes people will also put a style or class attribute or something on HR btw. But not in a simple description you'd hope.
